stdlib/messaging/rabbitmq/src/main/ballerina/src/rabbitmq/channel.bal

```diff
@@ -14,88 +14,129 @@
     public remote function basicPublish(@untainted MessageContent messageContent, string routingKey,
-                        string exchangeName = "", BasicProperties? properties = ()) returns Error? = external;
+                        string exchangeName = "", BasicProperties? properties = ()) returns Error? {
+        return nativeBasicPublish(messageContent, java:fromString(routingKey),
+         java:fromString(exchangeName), properties, self.amqpChannel);
+    }
 
     # Deletes the queue with the given name although it is in use or has messages on it.
-    # If the paramters ifUnused or ifEmpty is given, the queue is checked before deleting.
+    # If the parameters ifUnused or ifEmpty is given, the queue is checked before deleting.
     #
     # + queueName - Name of the queue to be deleted.
     # + ifUnused - True if the queue should be deleted only if not in use.
     # + ifEmpty - True if the queue should be deleted only if empty.
     # + return - An error if an I/O error is encountered or nil if successful.
-    public remote function queueDelete(string queueName, boolean? ifUnused = (), boolean? ifEmpty = ())
-                                            returns Error? = external;
+    public remote function queueDelete(string queueName, boolean ifUnused = false, boolean ifEmpty = false)
+                        returns Error? {
+        return nativeQueueDelete(java:fromString(queueName), ifUnused, ifEmpty, self.amqpChannel);
+    }
 
     # Deletes the exchange with the given name.
     #
     # + exchangeName - The name of the exchange.
     # + return - An I/O error if an error is encountered or nil otherwise.
-    public remote function exchangeDelete(string exchangeName) returns Error? = external;
+    public remote function exchangeDelete(string exchangeName) returns Error? {
+        return nativeExchangeDelete(java:fromString(exchangeName), self.amqpChannel);
+    }
 
     # Purges the contents of the given queue.
     #
     # + queueName - The name of the queue.
     # + return - An error if an I/O error is encountered or nil if successful.
-    public remote function queuePurge(string queueName) returns Error? = external;
+    public remote function queuePurge(string queueName) returns Error? {
+        return nativeQueuePurge(java:fromString(queueName), self.amqpChannel);
+    }
+
+    # Retrieves a message synchronously from the given queue, providing direct access to the messages in the queue.
+    #
+    # + queueName - The name of the queue.
+    # + ackMode - Type of acknowledgement mode.
+    # + return - `Message` object containing the retrieved message data or an `Error` if an
+    #               I/O problem is encountered.
+    public remote function basicGet(string queueName, AcknowledgementMode ackMode) returns Message | Error {
+        boolean autoAck = false;
+        if (ackMode is AUTO_ACK) {
+            autoAck = true;
+        }
```

Comment:
can use single line `boolean autoAck = ackMode is AUTO_ACK;`

Reply:
Fixed